_posts/2020-05-18-mirror-kafka-eventhub.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: post
+title:  "Mirroring from Apache Kafka to Azure Event Hub"
+date: 2020-05-18
+author: paolo_patierno
+---
+
+Let's imagine to have a software architecture based on Apache Kafka running on an on-premise Kubernetes installation and managed by the Strimzi operators.
+All the services are able to communicate with each other exchanging messages through a set of topics.
+At some point there is a new requirement to fullfil: all messages, sent to a specific topic, have to trigger some processing in the cloud via Microsoft Azure Functions already connected to an Azure Event Hub instance.
+It is a third party application already in place that we need to integrate with our own system.
+In order to avoid the producer applications to connect both the on-premise Apache Kafka cluster and to Azure Event Hub, which provides a Kafka protocol head, and sending each message twice, the best solution could be just mirroring the topic.
```

Comment:
```suggestion
In order to avoid the producing applications having to connect to both the on-premise Apache Kafka cluster and to Azure Event Hub, which provides a Kafka protocol head, and sending each message twice, the best solution could be just mirroring the topic.
```